lib/rubygems/commands/install_command.rb

```diff
@@ -202,10 +202,15 @@ def install_gem name, version # :nodoc:
       request_set = inst.resolve_dependencies name, req
 
       if options[:explain]
-        puts "Gems to install:"
+        say "Gems to install:"
 
         request_set.sorted_requests.each do |s|
-          puts "  #{s.full_name}"
+          # shows platform specific gems if used
+          if (plat = s.spec.platform) == Gem::Platform::RUBY
+            say "  #{s.full_name}"
```

Comment:
`Gem::BasicSpecification#full_name` already includes the platform like this

Reply:
No.  Two item.  If you comment out the added conditional logic, the following test fails:
```
  2) Failure:
TestGemCommandsInstallCommand#test_explain_platform_local [C:/Greg/GitHub/rubygems/test/rubygems/test_gem_commands_install_command.rb:1154]:
Expected: "  a-2-x86-mswin32-60"
  Actual: "  a-2"
```

`RequestSet#sorted_requests` returns an array of `Gem::Resolver::ActivationRequest`.
Their `full_name` method is `@spec.full_name`.  But `@spec` is a `Gem::Resolver::IndexSpecification`, which inherits it's method from `Gem::Resolver::Specification`, and its `full_name` method's comments state:

The name and version of the specification.
Unlike `Specification#full_name`, the platform is not included.

Reply:
should we fix that implementation, then, rather than reimplenting what "real" specs do?

Reply:
Your call.  Given that the comment mentions the difference, and it is not tagged :nodoc, is that a breaking change?

Reply:
Ah, didn't realize that was coming from the comment. In that case, maybe just extract into a method that, when given something that looks like a gem spec, does this dance, so we dont repeat it all over the place?

Reply:
This messy.  There are two places where similar code exists in install and update.

But, update is using an array, install is using a `Gem::Resolver::Specification`.  So refactoring would only change install.  Hence, maybe (for the time being) leave as is?

Off topic - Apologies, as these sort of things cloud my working in RG.  Its not new, and its been worked on by a lot of people.  IMHO, some things have gotten to the point where they're just messed up.  Hopefully this weekend I'll post an issue with some thoughts on that, and what I see as possible solutions...